core/unit_test/TestViewCopy.hpp

```diff
@@ -119,6 +119,43 @@ struct TestViewCopy {
     Kokkos::deep_copy( srcView, dstView );
     Kokkos::fence();
    }
+
+   // Test 3: deep_copy from host_space to ExecSpace and ExecSpace back to host_space
+  #if defined( KOKKOS_ENABLE_OPENMP ) || defined( KOKKOS_ENABLE_SERIAL ) || defined( KOKKOS_ENABLE_THREADS )
```

Comment:
This still doesn't do it for UVM memory!
We need to do 
CudaUVMSpace -> HostSpace 
HostSpace -> CudaUVMSpace
CudaHostPinnedSpace -> HostSpace
HostSpace -> CudaHostPinnedSpace
ROCmHostPinnedSpace -> HostSpace
HostSpace -> ROCmHostPinnedSpace 
...

Reply:
@crtrott In each test scope there are two `deep_copy` calls, where the views are interchanged for bi-directional part of the test. 
The actual test file is only included in the TestCudaUVMSpace_ViewCopy.cpp, TestCudaHostPinnedSpace_ViewCopy.cpp, and TestROCmHostPinnedSpace_ViewCopy.cpp files, each of which includes it's corresponding TestCuda*_Category.hpp. 
Each of these (memory) spaces is passed via template parameter `TEST_EXECSPACE` to `TestViewCopy` and used as the space when declaring the views required for testing here. I add `cout` statements with `typeid(view).name()` to check the memory spaces of the View's and they are doing what I thought was intended here, if not we can talk offline to clear up my misunderstanding.